curryreader.py

```diff
@@ -167,7 +167,7 @@ def read(inputfilename='', plotdata = 1, verbosity = 2):
            text = contents[ixstart:ixstop].strip()
            if text == 'ASCII' or text == 'CHAN' :   # test for alphanumeric values
                a[i] = 1
-           elif text.isnumeric() :
+           elif text.replace('.','',1).isnumeric() :  # BUG (float not recognized) FIXED
                a[i] = float(text)                   # assign if it was a number

```

Comment:
Since there's nothing else after this `elif` block, I think it's a bit cleaner to do the try-except method (even if it's a tad slower). This isn't a performance bottleneck (the loop only runs 16 times). Also the `.replace(...).isnumeric()` method won't handle negative numbers (probably that won't matter given the names of the fields above, they all appear to be strictly positive quantities, but better safe than sorry?)
```suggestion
           try:
               a[i] = float(text)
           except ValueError:
               pass
```